sound/soc/sof/loader.c

```diff
@@ -389,8 +390,14 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 	}
 
 	mutex_lock(&sdev->cores_mask_mutex);
-	/* fw boot is complete. Update the active cores mask */
+	/* FW boot is completed, update the status of dsp cores */
 	sdev->enabled_cores_mask = init_core_mask;
+	while (init_core_mask && (i < SND_SOF_CORE_MAX)) {
+		if (init_core_mask & BIT(0))
+			atomic_set(&sdev->core_refs[i], 1);
```

Comment:
you are already operating with the protection of the mutex, so why do you need an atomic_set?